### PR TITLE
updated head and tail docstrings

### LINKED ISSUE #46079: DOC: tail works even if n > length of DataFrame
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.tail.html

### Documentation problem

It may be obvious to some, but if the `n` requested by `tail()` is greater than the length of the DataFrame, you get the entire DataFrame and no errors are thrown, and this is not mentioned in the documentation. This is useful for me because I'm trying to plot only the last 70 data points of a dataframe but the dataframe might not have 70 values yet. I think it would be reasonable to assume that one implementation would throw an out-of-bounds style exception. 

### Suggested fix for documentation

Add one line about how n > dataframe length is handled.

### PR DESCRIPTION
**Pull request by INDIG0N**
- [ ] closes #46079

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -5284,6 +5284,8 @@ def head(self: NDFrameT, n: int = 5) -> NDFrameT:
         For negative values of `n`, this function returns all rows except
         the last `n` rows, equivalent to ``df[:-n]``.
 
+        If n is larger than the number of rows, this function returns al rows.
```
Comment (mroeschke) [inline pandas/core/generic.py:5287]: `al` -> `all`? Same as below

Comment (mroeschke) [reply pandas/core/generic.py:5287]: You can just add another commit fixing this typo to this pull request. When merged, all the commits will be squashed.

```
# Fix the typo
git add pandas/core/generic.py
git commit -m"Fixed all typo"
git push origin <branch_name>
```

**Comment by INDIG0N** [comment]
does anybody know why that one check is failing? I can't figure it out.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @INDIG0N the failures were unrelated

**Pull request merged**